### PR TITLE
Fixed normalisation in NormUnet to work when c != 2. Added optional `norm_per_coil` option.

### PR DESCRIPTION
**Pull request by Timsey**

### CODE REVIEW

**Review by mmuckley — CHANGES_REQUESTED** [review]
Could you clarify the use case a little bit more? We discussed earlier that the input to `norm` will always have a channel dimension of 2 due to the call of `complex_to_chan_dim`. Are you intending to call `norm` without this function?

A secondary item is that with the new parameter we should have a new unit test.

File: fastmri/models/varnet.py [hunk 1]
```diff
@@ -66,19 +71,42 @@ def chan_complex_to_last_dim(self, x: torch.Tensor) -> torch.Tensor:
...
+        if self.norm_per_coil:
+            return x * std + mean
+        else:
+            # x has shape [B, C, H, W], mean and std have shape [B, 2, 1],
+            #  so we need to shape x to [B, 2, C // 2 * H * W] before unnormalisation,
+            #  and then reshape back to [B, C, H, W].
+            b, c, h, w = x.shape
+            x = x.reshape(b, 2, c // 2 * h * w)
+            unnorm_x = x * std + mean
+            return unnorm_x.reshape(b, c, h, w)
```
[the remark this answers is not in the document]
Comment (Timsey) [reply fastmri/models/varnet.py:109]: I think so yeah, unless [B, 2, 1] can automatically be broadcast [B, C != 2, whatever].

**Comment by Timsey** [comment]
> Could you clarify the use case a little bit more? We discussed earlier that the input to `norm` will always have a channel dimension of 2 due to the call of `complex_to_chan_dim`. Are you intending to call `norm` without this function?
> 
> A secondary item is that with the new parameter we should have a new unit test.

Originally yes, we discussed some use cases where the input to NormUnet itself is multi-coil, which requires this change.

`complex_to_chan_dim` only guarantees c is a multiple of 2, which is fine with the current example, because the input to NormUnet is always functionally single-coil (both in the sensitivity module and the refinement module). 

Even if we only want to use functionally single-coil input though, the old version of the code was a bit confusing, because it didn't explicitly require that c == 2, but in fact would fail to broadcast correctly for c != 2.

**Comment by mmuckley** [comment]
@Timsey and I are closing this in favor of #163. For these features we may release a different model at some point.

**Pull request closed without merging**